packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -37,9 +37,63 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     double maxHeight,
     int imageQuality,
     CameraDevice preferredCameraDevice = CameraDevice.rear,
-  }) {
+  }) async {
     String capture = computeCaptureAttribute(source, preferredCameraDevice);
-    return pickFile(accept: _kAcceptImageMimeType, capture: capture);
+    return pickFile(
+      accept: _kAcceptImageMimeType,
+      capture: await _resizeImage(
+        capture,
+        maxWidth.toInt(),
+        maxHeight.toInt(),
+        imageQuality,
+      ),
+    );
+  }
+
+  static Future<String> _resizeImage(
+    String src,
+    int maxWidth,
+    int maxHeight,
+    int imageQuality,
+  ) {
+    final completer = Completer<String>();
+    final img = html.ImageElement();
+    img.onError.listen((event) {
+      completer.complete("");
+    });
+    img.onLoad.listen((event) {
+      final canvas = html.CanvasElement();
+      final ctx = canvas.context2D;
+      double ratio = 1;
+
+      if (img.width > img.height) {
+        if (img.width > maxWidth) ratio = maxWidth / img.width;
+      } else if (img.height > maxWidth) {
+        ratio = maxWidth / img.height;
+      }
+      if (img.width > img.height) {
+        if (img.width < maxHeight) ratio = maxHeight / img.width;
```

Comment:
This is odd, comparing a width to a height. I think there's some typos in this algo to determine the aspect ratio / scale factor (?)

Reply:
I'll take a look a the proper algorithm here, maybe I can refine this.

Reply:
@ditman , the scaling algorithm was just a copy paste from an old JavaScript project. I really don't recall how I constructed it or tested it thoroughly. 

The best algorithm for the resize function would be using the algorithm used in the mobile counter part. Maybe the algorithm may be copied from the following Java file so that it will be consistent with the mobile version.

https://github.com/flutter/plugins/blob/6783cd5ebd54e034a77e2404f5565b48bd19e121/packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java#L56

Reply:
I agree with you @postacik. I'll update the current algorith to resemble something similar to the Android implementation.